internal/build/buildconfig/manager.go

```diff
@@ -0,0 +1,187 @@
+package buildconfig
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	buildv1 "github.com/openshift/api/build/v1"
+	kmmv1beta1 "github.com/rh-ecosystem-edge/kernel-module-management/api/v1beta1"
+	"github.com/rh-ecosystem-edge/kernel-module-management/internal/auth"
+	"github.com/rh-ecosystem-edge/kernel-module-management/internal/build"
+	"github.com/rh-ecosystem-edge/kernel-module-management/internal/registry"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const builderServiceAccountName = "builder"
+
+var (
+	ErrNoMatchingBuildConfig = errors.New("no matching BuildConfig")
+	errNoMatchingBuild       = errors.New("no matching Build")
+)
+
+type buildConfigManager struct {
+	authFactory     auth.RegistryAuthGetterFactory
+	client          client.Client
+	coreClient      kubernetes.Interface
+	helper          build.Helper
+	maker           Maker
+	ocpBuildsHelper OpenShiftBuildsHelper
+	registry        registry.Registry
+}
+
+func NewManager(
+	client client.Client,
+	coreClient kubernetes.Interface,
```

Comment:
What is the difference between `client.Client` nad `kubeneretens.Interface`?

Reply:
@ybettan `kubernetes.Interface` is from `client-go` while `client.Client` is a generic overlay on top from controller-runtime.
I would not have our own client like we had in SRO, which would bring a lot of code. I would stick to `client.Client` as much as possible.